 public void testBugzilla214431() throws Exception  {  CDOSession session = openModel1Session();  CDOTransactionImpl transaction = (CDOTransactionImpl)session.openTransaction();  CDOResource resource = transaction.createResource("/test1");  SalesOrder salesOrder = getModel1Factory().createSalesOrder();  resource.getContents().add(salesOrder);  transaction.commit();  salesOrder.setId(4711);  assertNotSame(CDOUtil.getCDOObject(salesOrder).cdoRevision(), transaction.getRevision(CDOUtil.getCDOObject(  salesOrder).cdoID(), true));  assertEquals(CDOUtil.getCDOObject(salesOrder).cdoRevision(), transaction.getDirtyObjects().get(  CDOUtil.getCDOObject(salesOrder).cdoID()).cdoRevision());  transaction.close(); } 
